userspace/libsinsp/threadinfo.cpp

```diff
@@ -543,30 +551,44 @@ void sinsp_threadinfo::set_group(uint32_t gid)
 	}
 	if (group)
 	{
-		memcpy(&m_group, group, sizeof(scap_groupinfo));
+		m_group->set_gid(group->gid);
+
+		if (m_inspector->is_user_details_enabled())
+		{
+			m_group->set_name(group->name, sizeof(group->name));
+		}
 	}
 	else
 	{
-		m_group.gid = gid;
-		strlcpy(m_group.name, (gid == 0) ? "root" : "<NA>", sizeof(m_group.name));
+		// No need to set name/homedir/shell, the default values from
+		// sinsp_userinfo are going to be used.
+		m_group->set_gid(gid);
 	}
-	m_user.gid = m_group.gid;
+	m_user->set_gid(m_group->gid());
 }
 
 void sinsp_threadinfo::set_loginuser(uint32_t loginuid)
 {
 	scap_userinfo *login_user = m_inspector->m_usergroup_manager.get_user(m_container_id, loginuid);
+
 	if (login_user)
 	{
-		memcpy(&m_loginuser, login_user, sizeof(scap_userinfo));
+		m_loginuser->set_uid(login_user->uid);
+		m_loginuser->set_gid(m_group->gid());
+
+		if (m_inspector->is_user_details_enabled())
+		{
+			m_loginuser->set_name(login_user->name, sizeof(login_user->name));
+			m_loginuser->set_homedir(login_user->homedir, sizeof(login_user->homedir));
+			m_loginuser->set_shell(login_user->shell, sizeof(login_user->shell));
+		}
 	}
 	else
 	{
-		m_loginuser.uid = loginuid;
-		m_loginuser.gid = m_group.gid;
-		strlcpy(m_loginuser.name, loginuid == 0 ? "root" : "<NA>", sizeof(m_loginuser.name));
-		strlcpy(m_loginuser.homedir, loginuid == 0  ? "/root" : "<NA>", sizeof(m_loginuser.homedir));
-		strlcpy(m_loginuser.shell, "<NA>", sizeof(m_loginuser.shell));
+		// No need to set name/homedir/shell, the default values from
+		// sinsp_userinfo are going to be used.
+		m_loginuser->set_uid(loginuid);
+		m_loginuser->set_gid(m_group->gid());
```

Comment:
And again :)
```
strlcpy(m_loginuser.name, loginuid == 0 ? "root" : "<NA>", sizeof(m_loginuser.name));
strlcpy(m_loginuser.homedir, loginuid == 0  ? "/root" : "<NA>", sizeof(m_loginuser.homedir));
```

Reply:
Oh i see it now, it is buried inside helper methods. Forget these comments then!